bin/cli.js

```diff
@@ -20,30 +24,51 @@ handleInput(userAction, argv);
 function handleInput(action, argz) {
   switch (action) {
     case 'flash':
-      if (!argz.f || !argz.a) { 
+      if (!argz.f || !argz.a) {
         return showHelp();
         process.exit(1);
       }
-      else if (!boards[argz.a]) {
+      else if (!boards.byName[argz.a]) {
         console.error(new Error('Oops! That board is not supported, sorry.'));
         process.exit(1);
       }
       else {
+        // run flash function here if all is well
         var options = {};
         options.board = argz.a;
         if (argz.p) { options.port = argz.p; }
         if (argz.v) { options.debug = true; }
         flash(argz.f, options);
       }
-      // run flash function here if all is well
+      break;
+    case 'boards':
+      var boardNames = Object.keys( boards.byName ).sort();
+      console.log("Supported Boards:\n - " + boardNames.join("\n - ") );
+      break;
+    case 'list':
+      Avrgirl.listPorts(function(err,ports) {
+        console.log("Listing Devices:");
+        if ( argz.raw ) {
+          console.log( ports );
+        } else {
+          for (var i=0;i<ports.length;i++) {
+            var port = ports[i];
+            if ( port._standardPid ) {
+              console.log(
+                util.format(" - port: %s / pid: %s / mfg: %s / board: %s",
+                  port.comName,
+                  port._standardPid || 'unknown',
+                  port.manufacturer || 'unknown',
+                  port._boardNames.length > 0 ? port._boardNames.join(', ') : 'unknown'
+                )
+              );
+            }
+          }
+        }
+      });
       break;
     case 'help':
-      showHelp();
```

Comment:
These seemed like redundant calls to showHelp(). Since 

```
case 'help':
case undefined:
default:
```

all take the same code path, we can take out the breaks and have them fall through to the same call to showHelp. If this isn't your style, I can revert it.